x-pack/plugin/sql/src/main/java/org/elasticsearch/xpack/sql/planner/QueryFolder.java

```diff
@@ -455,6 +455,12 @@ static EsQueryExec fold(AggregateExec a, EsQueryExec exec) {
                 // literal
                 if (target.foldable()) {
                     queryC = queryC.addColumn(ne.toAttribute());
+                    if(a.groupings().isEmpty() == false){
```

Comment:
This if is redundant since below you iterate over the groupings, so if empty => no iteration